Source/Public/LevelStatsCollector.h

```diff
@@ -14,16 +28,33 @@ class MAPMETRICSGENERATION_API ALevelStatsCollector final : public AActor
 
     void PostInitializeComponents() override;
     void BeginPlay() override;
-    void Tick( float DeltaTime ) override;
+    void Tick( float delta_time ) override;
+
+    float GetMetricsWaitDelay() const;
```

Comment:
instead of exposing all these functions, you can make the state classes friends of the actor so that they can use its private members.
the states here are used to make the behavior more logically organized so its OK to befriend them

Reply:
done

Reply:
`so that they can use its private members.`

you can delete the functions you now made private, and move the code of these functions in each state

Reply:
done

Reply:
POST FORCEPUSH NOTE : I will clean the needed functions in the next PR with the JSON export when possible.